lib/pure/asyncdispatch.nim

```diff
@@ -355,27 +355,50 @@ proc `or`*[T, Y](fut1: Future[T], fut2: Future[Y]): Future[void] =
   fut2.callback = cb
   return retFuture
 
-proc all*[A](futs: seq[Future[A]]): Future[seq[A]] =
+proc all*[T](futs: varargs[Future[T]]): auto =
```

Comment:
Is there any way to specify the return type explicitly? ``auto`` sucks for documentation. If not, then at least document it please.